dotnet/sdk-phase-1.md

```diff
@@ -0,0 +1,1101 @@
+
+#if NETCOREAPP3_0_OR_GREATER
+    public async ValueTask DisposeAsync() { /*...*/ }
+#endif
+
+    internal class EphemeralServerBased : WorkflowEnvironment
+    {
+        public EphemeralServerBased(TemporalClient client) : base(client) { /*...*/ }
+    }
+}
+```
+
+Notes:
+
+* We call the method `ActivityEnvironment.Run` instead of `ActivityEnvironment.RunActivity` to make it clear that we
+  accept any code/lambda, it does not have to be an `[Activity` method
+* `WorkflowEnvironment` is `IAsyncDisposable` in all but the oldest .NET versions, which means it can be used like
+  `await using (var env = await WorkflowEnvironment.StartLocal()) { ...` (or just in the function without a block)
+* While it may seem unnecessary to build this now, it will be very helpful during tests of the framework itself. So we
+  should go ahead and build it
+
+## Runtime Support/Extensions
+
+.NET provides several
+[extensions to the runtime library](https://learn.microsoft.com/en-us/dotnet/standard/runtime-libraries-overview#extensions-to-the-runtime-libraries)
+which many users use when making apps. We discuss here some of those patterns and how we can support those, either by
+our existing design, or with helpers in a `Temporalio.Extensions` namespace.
+
+With DI and configuration and such, this is very analogous to Spring Boot for Java.
+
+### Serialization
+
+https://learn.microsoft.com/en-us/dotnet/standard/serialization/
+
+* We use `System.Text.Json` by default and therefore support all customization/approaches it provides
+* Our JSON payload converters may offer easy approaches to customize options to create new converters
+
+### Dependency Injection
+
+https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection
+
+* We have `Temporalio.Client.ITemporalClient` and `Temporalio.Client.ITemporalConnection` to allow DI for these impls
+* Should we also do this for `Temporalio.Client.WorkflowHandle`? Is there value in injecting that?
+* Since activity instances are instantiated by the user upon registration, the user can use DI when creating those
+  classes to, say, inject a logger
+  * We'll need an example showing this
+* When we start the workflow implementation side, we may need to revisit how DI may help users without forcing it
+
+### Configuration
+
+https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration
+
+* We support the https://learn.microsoft.com/en-us/dotnet/core/extensions/options pattern with all of our `Options`
+  classes
+  * Some of the options things like activities method references can't by loaded from a file, so we should take a look
+    at helpers for converting activity strings to activity classes reflectively
+  * We need a sample showing configuring workers via config file akin to Spring Boot
+* See the host section for how it all may come together inside a worker service
+
+### Logging
+
+https://learn.microsoft.com/en-us/dotnet/core/extensions/logging
```

Comment:
Speaking purely from an consumer point of view I think it would be a poor decision not to include this as the default logging implementation. Microsoft.Extensions.Logging is the de-facto logging mechanism in dotnet and has been for a while now and its interface seems to be very stable.

I think the vast majority of consumers would end up creating their own `DotnetTemporalLogger` implementation which all just bridge this gap. Without understanding the how ITemporalLogger would work you'd lose some of the freebies you get from using the standard `ILogger<>` implementation (like automatic class names, parameterized logs, etc.)

Reply:
This may have convinced me, I may go ahead and add a dependency on this package (as much as I absolutely abhor required dependencies for optional things like logging in a library like ours).